src/main/scala/io/iohk/ethereum/blockchain/sync/fast/FastSync.scala

```diff
@@ -874,25 +884,27 @@ class FastSync(
         requestReceipts(peer)
       } else if (syncState.blockBodiesQueue.nonEmpty) {
         requestBlockBodies(peer)
-      } else if (syncState.blockHeadersQueue.nonEmpty) {
+      } else if (blockHeadersQueue.nonEmpty) {
         requestBlockHeaders(peer)
       } else if (shouldRequestNewSkeleton(peerInfo)) {
         requestSkeletonHeaders(peer)
       } else {
-        log.debug("Nothing to request. Waiting for responses for [{}] sent requests.", assignedHandlers.size)
+        log.debug(
+          "Nothing to request. Waiting for responses for [{}] sent requests.",
+          assignedHandlers.size + skeletonHandler.size
+        )
       }
     }
 
     private def shouldRequestNewSkeleton(peerInfo: PeerInfo): Boolean =
       currentSkeletonState.isEmpty &&
         skeletonHandler.isEmpty &&
-        syncState.bestBlockHeaderNumber < syncState.safeDownloadTarget &&
-        peerInfo.maxBlockNumber >= syncState.pivotBlock.number
```

Comment:
need help understanding removing this line :)

Reply:
sure :) the check here didn't make a lot of sense because `peer` usually wasn't used for the download anyway (we use the master peer if it's defined).

it has been replaced with the logic here: https://github.com/input-output-hk/mantis/pull/951/files#diff-4b491aec4f7ba3bc8091f81d49100e0ccc38a2f396bcca38434118637e6c0befR987

which basically says:
- if we have a master peer and its best block is high enough, use the master peer
- otherwise check if this peer's best block is high enough and if yes, use it to download the header skeleton and as new master peer
- if the peer also doesn't have a high enough best block log a warning (logic will be called again with another peer automatically by the scheduled `processSyncing()` call)

Reply:
clear, thanks!